tests/ballerina-integration-test/src/test/resources/resiliency/resiliencyservices/14_http_load_balancer.bal

```diff
@@ -0,0 +1,260 @@
+            var responseToCaller = caller->respond(outResponse);
+            if (responseToCaller is error) {
+                log:printError("Error sending response", err = responseToCaller);
+            }
+        }
+    }
+}
+
+@http:ServiceConfig { basePath: "/mock1" }
+service mock1 bind backendEP {
+    @http:ResourceConfig {
+        path: "/"
+    }
+    mock1Resource(endpoint caller, http:Request req) {
+        var responseToCaller = caller->respond("Mock1 Resource is Invoked.");
+        if (responseToCaller is error) {
+            log:printError("Error sending response from mock service", err = responseToCaller);
+        }
+    }
+}
+
+@http:ServiceConfig { basePath: "/mock2" }
+service mock2 bind backendEP {
+    @http:ResourceConfig {
+        path: "/"
+    }
+    mock2Resource(endpoint caller, http:Request req) {
+        var responseToCaller = caller->respond("Mock2 Resource is Invoked.");
+        if (responseToCaller is error) {
+            log:printError("Error sending response from mock service", err = responseToCaller);
+        }
+    }
+}
+
+@http:ServiceConfig { basePath: "/mock3" }
+service mock3 bind backendEP {
+    @http:ResourceConfig {
+        path: "/"
+    }
+    mock3Resource(endpoint caller, http:Request req) {
+        var responseToCaller = caller->respond("Mock3 Resource is Invoked.");
+        if (responseToCaller is error) {
+            log:printError("Error sending response from mock service", err = responseToCaller);
+        }
+    }
+}
+
+@http:ServiceConfig { basePath: "/mock4" }
+service mock4 bind backendEP {
+    @http:ResourceConfig {
+        path: "/"
+    }
+    mock4Resource(endpoint caller, http:Request req) {
+        runtime:sleep(5000);
+        var responseToCaller = caller->respond("Mock4 Resource is Invoked.");
+        if (responseToCaller is error) {
+            log:printError("Error sending response from mock service", err = responseToCaller);
+        }
+    }
+}
+
+@http:ServiceConfig { basePath: "/mock5" }
+service mock5 bind backendEP {
+    @http:ResourceConfig {
+        path: "/"
+    }
+    mock5Resource(endpoint caller, http:Request req) {
+        runtime:sleep(5000);
+        var responseToCaller = caller->respond("Mock5 Resource is Invoked.");
+        if (responseToCaller is error) {
+            log:printError("Error sending response from mock service", err = responseToCaller);
+        }
+    }
+}
+
+# Implementation of custom load balancing strategy.
+#
+# + index - Keep tracks the current point of the CallerActions[]
+public type CustomLoadBalancerRule object {
+
+    public int index;
+
+    # Provides an HTTP client which is choosen according to the custom algorithm.
+    #
+    # + loadBalanceCallerActionsArray - Array of HTTP clients which needs to be load balanced
+    # + return - Choosen `CallerActions` from the algorithm or an `error` for a failure in
+    #            the algorithm implementation
```

Comment:
This comment should be related to method 'getNextCallerActions' defined in line 243. So shall we move this comment just before the method?

Reply:
Thanks, Actually documentation needs to go Just above the public method definition. Moved it to the correct place.